### ISSUE TITLE
General Discussion

### ISSUE THREAD

**Issue by ctyler** [pk400/SPO600-Build-Framework#1]
This "issue" is for general project discussions.

**Comment by playhan** [comment]
What kind of language do we use for this?

**Comment by playhan** [comment]
Will we set the rule of committing(pushing) ?
I mean the rule of writing comments when we commit.

**Comment by pk400** [comment]
I will manage the pull requests so having detailed comments will help a lot. After forking and have something you want to add to the main repo submit a pull request and I'll add it to the main repo if it doesn't break the framework.

**Comment by yehoshuajg** [comment]
@playhan Since we are using standard input and output for this project, you're pretty safe to use just about any language you feel comfortable with.

**Comment by ctyler** [comment]
I just sent a pull request for the first cut at the m10k top-level script.

A couple of notes:
- I coded it so that the modules will be found in ~/.m10k if it exists, otherwise it will look in /usr/local/m10k
- The script aborts on a bad build or bad test -- I wasn't sure if this was the behavior we wanted, or whether it should continue.
- Results guys: what format should I write build time and test results/time data in? Please define that somewhere and I'll add in the code.

**Comment by playhan** [comment]
Should we deal with exceptional handling for duplicated options?
For example,
(option group1) -alpha -beta -delta
(option group2) -seta -sigma -alpha

**Comment by pk400** [comment]
Status so far:
- [x] Config file
- [x] Parser
- [x] Permutation
- [x] Build
- [ ] Testing
- [ ] Benchmark
- [ ] Analysis

If the Testing, Benchmarking, Analysis guys need any help don't hesitate to ask. More than halfway done.

**Comment by jjaku** [comment]
Where are the results stored by the benchmark runs? I found the .db file in ~/.m10k. Is that the only output?

**Comment by Protoster** [comment]
Yes, the results are stored where you mentioned. There should also be an
svg image with a chart. The db file is a sqlite database you can query
